src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -396,6 +403,8 @@ public int[] getDefaultTokens() {
             TokenTypes.BNOT,
             TokenTypes.POST_INC,
             TokenTypes.POST_DEC,
+            TokenTypes.INDEX_OP,
+            TokenTypes.DOT,
```

Comment:
whole issue is about extra parentheses before "." can we skip "INDEX_OP" in default tokens ?
and we can do extra effort to check INDEX_OP after we are called(visitToken) on  DOT.

I know list of tokens is confusing, but what we try to keep them for is for adjustments of targets that use might opt-in.

If all is about DOT , but INDEX_OP and LITERAL_NEW are good for good DOT support. it means tokens should have only DOT and java code in Check search for INDEX_OP and LITERAL_NEW under/around DOT, if DOT is detected.